src/floom/obstacle.js

```diff
@@ -1,12 +1,23 @@
 import Vector2 from "./../external/vector2.js";
 
-	var Obstacle = function(x, y, radius) {
-		this.position = new Vector2(x, y);
-		this.radius = radius;
+	var Obstacle = function(options) {
+		this.type = options.type;
+		if (options.type === 'circle') {
+			this.position = new Vector2(options.position.x, options.position.y);
+			this.radius = options.radius;
+		}
+		if (options.type === 'capsul') {
+			this.position = {
+				start: new Vector2(options.position.start.x, options.position.start.y),
+				end: new Vector2(options.position.end.x, options.position.end.y),
+			};
+			this.radius = options.radius;
+		}
 	};
 
 	Obstacle.prototype.toJSON = function() {
 		let settings = {
+			type: this.type,
 			position: this.position,
```

Comment:
`toJSON` has become rather complex: `position` is currently stored by reference, potentially having unintended side effects when modifying a previously stored `Obstacle`. Serialization should account for that (e.g. by employing a composite).

Reply:
simplified to direct